include/libswiftnav/logging.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2015 Swift Navigation Inc.
+ * Contact: Fergus Noble <fergus@swift-nav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef LIBSWIFTNAV_LOGGING_H
+#define LIBSWIFTNAV_LOGGING_H
+
+/* DEBUG off by default, enable it on a per-file basis. */
+#ifndef DEBUG
+#define DEBUG 0
+#endif
+
+/** \defgroup logging Logging
+ * Logging
+ *
+ * Logging at the `DEBUG` level is turned off by default and should be enabled
+ * on a per-file basis by adding the following line to the source file *before*
+ * including `logging.h`:
+ *
+ *    #define DEBUG 1
+ *
+ * \{ */
+
+/** Log an error.
+ * \param args `printf` style format and arguments.
+ */
+#define log_error(args...) printf("ERROR: " args)
+
```

Comment:
Usually I would agree, but I think there is a good chance they may not be macros in the final implementation...